deeplearning/ml4pl/graphs/unlabelled/llvm2graph/node_encoder.py

```diff
@@ -76,12 +76,12 @@ def EncodeNodes(self, g: nx.DiGraph) -> None:
       if text:
         data["preprocessed_text"] = text
         data["type"] = programl_pb2.Node.STATEMENT
-        data["x"] = [self.dictionary.get(data["text"], self.dictionary["!UNK"])]
+        data["x"] = [self.dictionary.get(text, self.dictionary["!UNK"])]
         data["y"] = []
       else:
         data["preprocessed_text"] = "!UNK"
         data["type"] = programl_pb2.Node.STATEMENT
-        data["x"] = [self.dictionary.get(data["text"], self.dictionary["!UNK"])]
```

Comment:
I'm not following, here `text` will be `""`.
That will likely still return the token ID of !UNK, but I think it seems principled to say that once we decide the preprocessed text is `!UNK` then that's what we should get the token for.

I suggest `data['preprocessed_text'] here.

Reply:
You're right, thanks! I've made the change in github.com/ChrisCummins/phd/pull/66 , which moves the node_encoder module